src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGANaivelySpark.java

```diff
@@ -0,0 +1,485 @@
+        results.cache(); // cache because Spark doesn't have an efficient RDD.split(predicate) yet
+        final JavaPairRDD<Long, SGAAssemblyResultType> success = results.filter(entry -> entry._1()!=null);
+        success.mapToPair(entry -> new Tuple2<>(entry._1(), new ContigsCollection(entry._2().assembledContigFile)))
+                .saveAsObjectFile(outputDir);
+        final JavaPairRDD<Long, SGAAssemblyResultType> failure = results.filter(entry -> entry._2()==null);
+        failure.mapToPair(entry -> new Tuple2<>(entry._1(), entry._2().runtimeInformation))
+                .saveAsObjectFile(outputDir);
+    }
+
+    /**
+     * Converts an entry in an JavaPairRDD whose first is an breakpoint id and second is an iterable of paired reads
+     *    where either end is "mapped" (which ever way the read picker think mapping means) to a place near
+     *    the breakpoint represented by the id.
+     * @param orderedReadsOfABreakPoint  the entry to be converted
+     * @return                           a tuple corresponding to this breakpoint where the first is still the breakpoint id,
+     *                                   and the second is another tuple of the temporary directory and a temporary FASTQ file
+     *                                   containing sequence information.
+     * @throws IOException               an IOException is thrown if the conversion fails to create the FASTQ file
+     */
+    @VisibleForTesting
+    static Tuple2<Long, File> convertReadsToFASTQ(final Tuple2<Long, Iterable<GATKRead>> orderedReadsOfABreakPoint) throws IOException{
+
+        final Long breakpointId = orderedReadsOfABreakPoint._1();
+
+        File workingDir = new File( Files.createTempDirectory("assembly" + orderedReadsOfABreakPoint._1().toString()).toString() );
+        if(null==workingDir){
+            throw new IOException("Failed to create temporary directory to perform assembly in.");
+        }
+        IOUtils.deleteRecursivelyOnExit(workingDir);
+
+        final File fastq = new File(workingDir, "assembly" + breakpointId + ".fastq");
+        if(null==fastq){
+            throw new IOException("Failed to create temporary FASTQ file to feed to SGA.");
+        }
+
+        try(final FastqWriter writer = new FastqWriterFactory().newWriter(fastq)){
+            Iterator<GATKRead> it = orderedReadsOfABreakPoint._2().iterator();
+            while(it.hasNext()){
+                writeToTempFASTQ(it.next(), writer);
+                writeToTempFASTQ(it.next(), writer);
+            }
+        }
+
+        return new Tuple2<>(breakpointId, fastq);
+    }
+
+    // copied from FindSVBreakpointsSpark
+    private static void writeToTempFASTQ(final GATKRead read, final FastqWriter writer){
+        final String readName      = read.getName();// TODO: hack for unit testing, where read names have /1 and /2. should be:   + (read.isSecondOfPair() ? "/2" : "/1");
```

Comment:
I'm not sure what the comment means -- why not add the /1 and /2 here? Doesn't SGA want/need them?